app/src/test/java/io/intrepid/contest/screens/entrysubmission/entryname/EntryNamePresenterTest.java

```diff
@@ -0,0 +1,64 @@
+package io.intrepid.contest.screens.entrysubmission.entryname;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import io.intrepid.contest.screens.entrysubmission.entryname.EntryNameContract.Presenter;
+import io.intrepid.contest.screens.entrysubmission.entryname.EntryNameContract.View;
+import io.intrepid.contest.testutils.BasePresenterTest;
+import io.intrepid.contest.testutils.TestPresenterConfiguration;
+
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EntryNamePresenterTest extends BasePresenterTest<EntryNamePresenter> {
+    @Mock
+    View mockView;
+
+    private Presenter presenter;
+
+    @Before
+    public void setup() {
+        presenter = new EntryNamePresenter(mockView, TestPresenterConfiguration.createTestConfiguration());
+    }
+
+    @Test
+    public void contestantWelcomeTextViewShouldShowContestName() {
+        presenter.onViewCreated();
+        verify(mockView).onContestNameRetrieved("Chili Cookoff");
+    }
+
+    @Test
+    public void entryNameTextViewFocusedShouldShowNextButton() {
+        presenter.onEntryNameTextViewFocusChanged(true);
+        verify(mockView).showEntryNameNextButton();
+        verify(mockView, never()).hideEntryNameNextButton();
+    }
+
+    @Test
+    public void entryNameTextViewFocusLostShouldHideNextButton() {
+        presenter.onEntryNameTextViewFocusChanged(false);
+        verify(mockView).hideEntryNameNextButton();
+        verify(mockView, never()).showEntryNameNextButton();
+    }
+
+    @Test
+    public void entryNameTextViewEmptyShouldDisableNextButton() {
+        CharSequence newText = "";
```

Comment:
Thank you! This is very useful, I didn't know about it.
In this case I "arranged" very explicitly for clarification. Do you think in most cases it's unnecessary to create a variable just for that? Or since it's a test it's good?